### PR TITLE
Update pyproject dependencies and add test to ensure it builds on conda-forge

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

This PR replaces https://github.com/nebari-dev/nebari/pull/1658.

This PR updates the pins in the `pyproject.toml` and adds a test that will ensure that the `nebari` package can be built and packaged by conda-forge. 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

### CODE REVIEW

**Comment by trallard** [comment]
Is this ready for review?

File: tests/test_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+        # get the directory the meta.yaml is in
+        meta_loc = tmp_path / "nebari"
+        # try to run conda build to build package from meta.yaml
+        subprocess.run(
+            ["conda", "build", meta_loc],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=True,
+        )
```
Comment (viniciusdc) [inline tests/test_dependencies.py:62]: Even though we are redirecting the execution to the terminal, I could see the build logs. Could you add a step to write the build logs to a file and cat those to the workflow? -- I will follow up cause I think conda-build already writes up this to a file by default

**Review by viniciusdc — COMMENTED** [review]
@iameskild Amazing work, all looks good. I added a quick comment around the fact that I couldn't see the build logs in the CI, which would be a good thing to debug if the errors from conda-build will be caught as an exception or not

File: tests/test_dependencies.py (hunk 1 above)
Comment (iameskild) [reply tests/test_dependencies.py:62]: @viniciusdc thanks for the review :)

I took your suggestion and made sure that the stderr gets printed when the test fails 👍

**Comment by viniciusdc** [comment]
Thanks @iameskild I will have another look over this tomorrow, but I am sure we will be able to merge it

**Review by viniciusdc — APPROVED** [review]
All working, amazing work @iameskild though I would like to inspect the build artifacts if possible (like the tar file, Grayskull recipe, and logs) this can be added later.  For now this looks like an incredible addition to help us keep track of the syncing scenario on PyPi and conda-forge.

**Pull request merged**